services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go

```diff
@@ -25,231 +26,421 @@ func init() {
-				logger.Error(
-					"Failed to hash data. Profile CUID: "+profile.Cuid,
-					err,
-				)
-				cleanUp()
-			}
+	// Save node_id to profile
+	if err := r.svc.UpdateNodeID(profile.Oid, nodeID); err != nil {
+		return fmt.Errorf(
+			"update node id failed. Profile ID: %s, error: %v",
+			profile.Oid,
+			err,
+		)
+	}
 
-			if profileHash != profile.SourceDataHash {
-				logger.Info(
-					"Source data hash mismatch: " + profile.Cuid + " - " + profile.Oid + " : " + profile.SourceDataHash + " - " + profileHash,
-				)
-
-				// reconstruct data
-				profileJSON, err = importutil.MapProfile(
-					profileData[0].(map[string]interface{}),
-					mapping,
-					schemaName,
-				)
-				if err != nil {
-					logger.Error(
-						"Map profile failed. Profile ID: "+profile.Oid,
-						err,
-					)
-					cleanUp()
-				}
-				oid := profileJSON["oid"].(string)
-
-				if profileJSON["primary_url"] == nil {
-					logger.Info("The primary_url is empty. Profile ID: " + oid)
-					continue
-				}
-
-				// validate data
-				validateURL := config.Conf.Index.URL + "/v2/validate"
-				isValid, failureReasons, err := importutil.Validate(
-					validateURL,
-					profileJSON,
-				)
-				if err != nil {
-					logger.Error(
-						"Validate profile failed. Profile ID: "+profile.Oid+". error message: ",
-						err,
-					)
-					cleanUp()
-				}
-				if !isValid {
-					logger.Info(
-						"Validate profile failed. Profile ID: " + profile.Oid + ". failure reasons: " + failureReasons,
-					)
-					cleanUp()
-				}
-				profileSvc := service.NewProfileService(
-					mongo.NewProfileRepository(mongodb.Client.GetClient()),
-				)
-				// save to Mongo
-				count, err := profileSvc.Count(profile.Oid)
-				if err != nil {
-					logger.Error(
-						"Can't count profile. Profile ID: "+profile.Oid,
-						err,
-					)
-					cleanUp()
-				}
-				if count <= 0 {
-					profileJSON["cuid"] = cuid.New()
-					err := profileSvc.Add(profileJSON)
-					if err != nil {
-						logger.Error(
-							"Can't add a profile. Profile ID: "+profile.Oid,
-							err,
-						)
-						cleanUp()
-					}
-				} else {
-					result, err := profileSvc.Update(profile.Oid, profileJSON)
-					if err != nil {
-						logger.Error("Can't update a profile. Profile ID: "+profile.Oid, err)
-						cleanUp()
-					}
-					profileJSON["cuid"] = result["cuid"]
-				}
-
-				// post update to Index
-				postNodeURL := config.Conf.Index.URL + "/v2/nodes"
-				profileURL := config.Conf.DataProxy.URL + "/v1/profiles/" + profileJSON["cuid"].(string)
-				nodeID, err := importutil.PostIndex(postNodeURL, profileURL)
-				if err != nil {
-					logger.Error(
-						"Failed to post profile to Index. Profile URL: "+profileURL,
-						err,
-					)
-					cleanUp()
-				}
-
-				// save node_id to profile
-				err = profileSvc.UpdateNodeID(oid, nodeID)
-				if err != nil {
-					logger.Error("Update node id failed. Profile ID: "+oid, err)
-					cleanUp()
-				}
-			} else {
-				err = svc.UpdateAccessTime(profile.Oid)
```

Comment:
@geoffturk @maxsarratt 

I've observed that the logic in this section has been removed, and I'd like to verify if this deletion is intentional and correct.
The logic here is that when there are no updates to the profile, we should update the access_time. However, I believe we've removed this practice.
If you can confirm that this is indeed the case, please indicate your agreement with a thumbs-up 👍.

Reply:
We should not delete this access time update. It is done to show in the metadata when was the last time we checked the profile data source for any changes to the profile. Even if nothing changed, we still want to update this timestamp. It is different from the `last_updated` timestamp that we store in every profile in the index.

Reply:
The logic resides in the `updateProfileAccessTime` method. When hashedData is not equal to profile.SourceDataHash, we will call this method.